spring-cloud-stream-binders/spring-cloud-stream-binder-kafka/src/test/java/org/springframework/cloud/stream/binder/kafka/KafkaBinderTests.java

```diff
@@ -107,6 +116,19 @@ public void init() {
 		}
 	}
 
+	@After
+	public void cleanup() {
+		try {
+			Seq<String> topics = ZkUtils.getAllTopics(kafkaTestSupport.getZkClient());
+			for (String topic: scala.collection.JavaConversions.seqAsJavaList(topics)) {
+				AdminUtils.deleteTopic(kafkaTestSupport.getZkClient(), topic);
+			}
+		}
+		catch (Exception e) {
+			log.error(e.getMessage());
```

Comment:
I believe it is still worth to keep the cleanup code just to avoid any overlap in future (given tests run sequentially). Also, it would be good to keep the `AbstractBinderTests` as is (with the embedded mode) and revisit if we need to configure for external broker support. I agree that the KafkaTestSupport code requires some cleanup related to the external broker support.